lib/merch_calendar/merch_week.rb

```diff
@@ -81,108 +93,96 @@ def year_week
+        calendar.end_of_month(merch_year, num) >= date && date >= calendar.start_of_month(merch_year, num)
       end
     end
 
-    # The merch year
+    # Returns the Merch year depending whether it is from the Retail or Fiscal calendar
     #
     # @return [Fixnum]
     def year
-      start_of_year.year
+      @year ||= calendar.merch_year_from_date(date)
     end
 
-    # The julian month that this merch week falls in
+    # Returns julian month where the merch week falls
     #
     # @return [Fixnum]
     def month
-      @month ||= MerchCalendar.merch_to_julian(merch_month)
+      @month ||= calendar.merch_to_julian(merch_month)
     end
 
-    # The specific quarter this week falls in
+    # Returns the quarter that this merch week falls
     #
     # @return [Fixnum]
     def quarter
-      case merch_month
-      when 7,8,9
-        return 1
-      when 10,11,12
-        return 2
-      when 1,2,3
-        return 3
-      else
-        return 4
-      end
+      @quarter ||= calendar.quarter(merch_month)
     end
 
-    # Returns the date of the start of this week
+    # Returns the start date of this week
     #
     # @return [Date]
     def start_of_week
       @start_of_week ||= (start_of_month + (7 * (week - 1)))
     end
 
-    # Returns the date of the end of this week
+    # Returns the end date of this week
     #
     # @return [Date]
     def end_of_week
       @end_of_week ||= (start_of_week + 6)
     end
 
-    # the number of the week within the given month
+    # The number of the week within the given merch month
     # will be between 1 and 5
     #
     # @return [Fixnum]
     def week
       @week ||= (((date-start_of_month)+1)/7.0).ceil
     end
 
-    # The date of the start of the corresponding merch year
+    # The start date of the corresponding merch year
     #
     # @return [Date]
     def start_of_year
-      @start_of_year ||= year_start_date
+      @start_of_year ||= calendar.start_of_year(year)
     end
 
     # The end date of the corresponding merch year
     #
     # @return [Date]
     def end_of_year
-      @end_of_year ||= retail_calendar.end_of_year(year)
+      @end_of_year ||= calendar.end_of_year(year)
     end
 
     # The start date of the merch month
     #
     # @return [Date]
     def start_of_month
-      @start_of_month ||= retail_calendar.start_of_month(year, merch_month)
+      @start_of_month ||= calendar.start_of_month(year, merch_month)
     end
 
     # The end date of the merch month
     #
     # @return [Date]
     def end_of_month
-      @end_of_month ||= retail_calendar.end_of_month(year, merch_month)
+      @end_of_month ||= calendar.end_of_month(year, merch_month)
     end
 
     # The merch season this date falls under.
     # Returns a string of +Fall/Winter+ or +Spring/Summer+
-    #
+    # THIS MIGHT NEED TO CHANGE DEPENDING ON THE CALENDAR
```

Comment:
Does this comment apply?  You seem to be changing it depending on the calendar.

Reply:
Deleted! 👍